mlabs/src/Mlabs/EfficientNFT/Marketplace.hs

```diff
@@ -9,45 +9,71 @@ import PlutusTx.Prelude
 import Ledger (
   CurrencySymbol,
   ScriptContext,
+  TokenName,
   TxInInfo (txInInfoResolved),
-  TxInfo (txInfoMint),
-  findOwnInput,
+  TxInfo (txInfoInputs, txInfoMint),
+  TxOut (txOutAddress),
+  Value,
   mkValidatorScript,
+  ownHash,
   scriptContextTxInfo,
+  scriptHashAddress,
   txOutValue,
  )
 import Ledger.Typed.Scripts (Any, TypedValidator, unsafeMkTypedValidator, wrapValidator)
-import Ledger.Value qualified as Value
+import Ledger.Value (getValue, valueOf)
+import Plutus.V1.Ledger.Ada (adaSymbol)
+import PlutusTx.AssocMap qualified as AssocMap
 
 -- | An escrow-like validator, that holds an NFT until sold or pulled out
 {-# INLINEABLE mkValidator #-}
-mkValidator :: CurrencySymbol -> BuiltinData -> BuiltinData -> ScriptContext -> Bool
-mkValidator nftCS _ _ ctx =
-  traceIfFalse "Tokens can only be redeemed when the policy allows a remint" checkRemint
-    && traceIfFalse "Inputs with more than one token are invalid" checkInputUTxO
+mkValidator :: BuiltinData -> BuiltinData -> ScriptContext -> Bool
+mkValidator _ _ ctx =
```

Comment:
this should just hold the token (AssetClass) to check for remint in the datum and if it does, validate. much simpler that way.